web/app/view/DeviceDialog.js

```diff
@@ -57,6 +57,22 @@ Ext.define('Traccar.view.DeviceDialog', {
             xtype: 'textfield',
             name: 'contact',
             fieldLabel: Strings.deviceContact
+        }, {
+            xtype: 'combobox',
+            name: 'category',
+            fieldLabel: Strings.deviceCategory,
+            store: 'DeviceImages',
+            queryMode: 'local',
+            displayField: 'name',
+            valueField: 'key',
+            listConfig: {
+                getInnerTpl: function () {
+                    return '<table><tr valign="middle" ><td><div align="center" style="width:40px;height:40px;" >' +
```

Comment:
I think it looks good enough for now... we can improve it in future if there are any issues.